op-deployer/pkg/deployer/pipeline/implementations.go

```diff
@@ -35,48 +31,24 @@ func DeployImplementations(ctx context.Context, env *Env, bundle ArtifactsBundle
 		contractsRelease = "dev"
 	}
 
-	var dump *foundry.ForgeAllocs
-	var dio opcm.DeployImplementationsOutput
-	err = CallScriptBroadcast(
-		ctx,
-		CallScriptBroadcastOpts{
-			L1ChainID:   big.NewInt(int64(intent.L1ChainID)),
-			Logger:      lgr,
-			ArtifactsFS: bundle.L1,
-			Deployer:    env.Deployer,
-			Signer:      env.Signer,
-			Client:      env.L1Client,
-			Broadcaster: KeyedBroadcaster,
-			Handler: func(host *script.Host) error {
-				host.ImportState(st.SuperchainDeployment.StateDump)
+	env.L1ScriptHost.ImportState(st.L1StateDump.Data)
```

Comment:
I'd like to understand this more.

Am I reading this correctly? 
We are building up forge allocs in `st.L1StateDump.Data` throughout the cli pipeline then using the `Host` ('an EVM executor that runs Forge scripts.') to import the state for use in the next steps of the pipeline? 
Gradually building all the necessary forge allocs in this `st.L1StateDump` field allows us to eventually be confident that we can great an L1 genesis out of them.

Reply:
Correct! We need to import it in each step in case the pipeline is interrupted partway through, and the data from the state on disk needs to be read into the host.